File: api/ingredient.py
from flask import Blueprint, jsonify, json, request
from document.ingredientMacros import IngredientMacros

ingredient_bp = Blueprint("ingredient", __name__, url_prefix="/api/ingredient")

@ingredient_bp.route("/add")
def add_ingredient():
    name = request.args.get("name")
    measurement = request.args.get("measurement")
    carbs = float(request.args.get("carbs"))
    fat = float(request.args.get("fat"))
    protein = float(request.args.get("protein"))
    cal = float(request.args.get("cal"))
    ingredient_macros = IngredientMacros(
        name=name,
        measurement=measurement,
        carbs=carbs,
        fat=fat,
        protein=protein,
        cal=cal
    )
    ingredient_macros.save()
    return 'added'


@ingredient_bp.route("/get")
def get_ingredient():
    ingredients = []
    for ingredient in IngredientMacros.objects:
        ingredient_json = ingredient.to_json()
        ingredient_data = json.loads(ingredient_json)
        ingredients.append(ingredient_data)
    return jsonify(ingredients)

File: frontend/scripts/page/Ingredient.js
import React, { useState, useEffect } from "react";


export default function Ingredient() {
    const [ingredients, setIngredients] = useState([])
    useEffect(() => {
		fetch("http://localhost:5000/api/ingredient/get")
		  .then(res => res.json())
		  .then(
			(result) => {
			  console.log(result)
			  setIngredients(result)
			},
			(error) => {
			  console.log(error)
			}
		  )
    }, [])
    if (ingredients.length == 0) {
    	return  <h3>Loading</h3>
    } else {
    	const ingredientList = ingredients.map((ingredient) =>
    	<tr>
    	  <td>{ingredient.name}</td>
		  <td>{ingredient.carbs}</td>
		  <td>{ingredient.fat}</td>
		  <td>{ingredient.protein}</td>
		  <td>{ingredient.cal}</td>
		 </tr>
		  );
		return <table>
		  <tr>
		    <th>Name</th>
		    <th>Carbs</th>
		    <th>Fat</th>
		    <th>Protein</th>
		    <th>Cal</th>
		  </tr>
		  {ingredientList}

		</table>
    }

}